File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = len(prices)
        lp,rp = 0,1
        max_profit = 0
        while rp < l:
            if prices[rp] > prices[lp]:
                profit = prices[rp] - prices[lp]
                max_profit = max(max_profit,profit)
            else:
                lp = rp
            rp=rp+1
        
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int small = p[0], maxP = 0;
        for(int i : p){
            small = min(small, i);
            maxP = max(maxP, i-small);
        }  
        return maxP;  
    }
};

